test/rss.cpp

```diff
@@ -489,6 +491,48 @@ TEST_CASE("RssFeed::sort() correctly sorts articles", "[rss]")
 		REQUIRE(articles[3]->pubDate_timestamp() == 23);
 		REQUIRE(articles[4]->pubDate_timestamp() == 7);
 	}
+
+	SECTION("random")
+	{
+		ArticleSortStrategy ss;
+		ss.sm = ArtSortMethod::RANDOM;
+
+		class G
+		{
+			private:
+				uint32_t i;
+			public:
+				typedef uint32_t result_type;
+
+				result_type min()
+				{
+					return 1;
+				}
+
+				result_type max()
+				{
+					return 5;
+				}
+
+				result_type operator()()
```

Comment:
It took me a minute to understand why this passed all tests even though `i` is never initialized. My understanding is this:
1. `i` gets initialized with whatever garbage was in memory at the time when `i` was allocated;
2. when this operator is called, there is an extremely high probability ((2^32-6)/(2^32), 99.99999986030161%) that `i` is outside of [0; 5] range. Thus the `if` is almost always executed, `i` is initialized to 1, and the following calls behave deterministically (i.e. always the same).

Please initialize `i` to something to eliminate that 0.00000013969839% chance of failure :)

Reply:
Regarding `i` I thought that private members are initialized to zero or something. I should've gone with a known quantity.